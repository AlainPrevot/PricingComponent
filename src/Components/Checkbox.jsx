import '../Sass/Checkbox.sass';
import PropTypes from 'prop-types';

export const Checkbox = ({ onMode }) => {
	return (
		<div className='container'>
			<span>Annually</span>
			<div className='checkbox'>
				<input type='checkbox' id='checkbox' onChange={() => onMode()} />
				<label htmlFor='checkbox'></label>
			</div>
			<span>Monthly</span>
		</div>
	);
};

Checkbox.propTypes = {
	onMode: PropTypes.func.isRequired,
};
