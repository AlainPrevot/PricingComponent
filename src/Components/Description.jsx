import PropTypes from 'prop-types'
import '../Sass/Description.sass'

export const Description = ({Storage, Users, Memory}) => {
  return (
    <div className='Card--Description border-top-bottom'>
        <span>{Storage} TB Storage</span>
        <span className='border-top-bottom'>{Users} Users Allowed</span>
        <span>Send up to {Memory} GB</span>
    </div>
  )
};

Description.proptypes = {
    Storage: PropTypes.number.isRequired,
    Users: PropTypes.number.isRequired,
    Memory: PropTypes.number.isRequired,
};

