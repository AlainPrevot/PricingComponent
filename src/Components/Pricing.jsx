import '../Sass/Pricing.sass';
import PropTypes from 'prop-types';

export const Pricing = ({ value }) => {
	return (
		<div className='card--pricing'>
			<span>$</span>
			<h1>{value}</h1>
		</div>
	);
};

Pricing.propTypes = {
	value: PropTypes.number.isRequired,
};
