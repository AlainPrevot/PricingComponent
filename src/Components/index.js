export * from './Card';
export * from './Checkbox';
export * from './ContainerCard';
export * from './Description';
export * from './Pricing';
export * from './Title';
export * from './TypePlan';