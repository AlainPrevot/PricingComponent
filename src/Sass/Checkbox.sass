.checkbox
    display: inline-block
    position: relative
    width: 56px
    height: 32px
    background: linear-gradient(145deg, hsl(236, 72%, 79%), hsl(237, 63%, 64%))
    border-radius: 20px
    margin: 0 auto

    input
        visibility: hidden

.checkbox:hover
    filter: brightness(1.4)
    box-shadow: none

.checkbox label
    /* aspecto */
    display: block
    width: 24px
    height: 24px
    border-radius: 50%
    box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35)
    background: #fff
    cursor: pointer

    /* Posicionamiento */
    position: absolute
    top: 4px
    left: 28px
    z-index: 1

    /* Comportamiento */
    transition: all .4s ease

    /* ocultar el posible texto que tenga */
    transition: left 500ms

.checkbox input[type=checkbox]:checked + label
    left: auto
    // right: 4px
    left: 4px

.container--checkbox
    display: flex
    width: 24%
    justify-content: space-between
    align-items: center
    color: hsl(232, 13%, 50%)

@media only screen and (max-width:1000px)

    .container--checkbox
        width: 80%
        margin-bottom: 80px
