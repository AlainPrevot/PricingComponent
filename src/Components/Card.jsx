import PropTypes from 'prop-types';
import { Description } from './Description';
import { Pricing } from './Pricing';
import '../Sass/Card.sass';
import { usePlan } from '../Hooks/usePlan';

export const Card = ({
	title,
	pricing,
	storage,
	users,
	memory,
	theme,
	plane,
}) => {
	const { newPricing } = usePlan(pricing, plane);

	return (
		<div className='card' data-theme={theme}>
			<h2>{title}</h2>
			<Pricing value={newPricing} />
			<Description Storage={storage} Memory={memory} Users={users} />
			<button className='card--button'>LEARN MORE</button>
		</div>
	);
};

Card.propTypes = {
	title: PropTypes.string.isRequired,
	pricing: PropTypes.number.isRequired,
	storage: PropTypes.number.isRequired,
	memory: PropTypes.number.isRequired,
	users: PropTypes.number.isRequired,
	theme: PropTypes.string.isRequired,
	plane: PropTypes.string.isRequired,
};
