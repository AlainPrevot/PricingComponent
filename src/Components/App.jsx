import { Card } from './Card';
import '../Sass/App.sass';
import { Title } from './Title';
import { Checkbox } from './Checkbox';
import { useState } from 'react';

export const App = () => {
	const [plan, setPlan] = useState('monthly');

	const TypePlan = () => {
		setPlan(plan === 'monthly' ? 'Annually' : 'monthly');
	};

	return (
		<div
			style={{
				display: 'flex',
				alignItems: 'center',
				height: '76vh',
				// justifyContent: 'space-around',
				flexDirection: 'column',
			}}
			className='cuerpo'
		>
			<Title title={'Our Pricing'} />
			<Checkbox onMode={TypePlan} />
			<div className='prueba'>
				<Card
					title={'Basic'}
					pricing={19.99}
					storage={500}
					memory={3}
					users={2}
					theme={'secundary'}
					plane={plan}
				/>
				<Card
					title={'Professional'}
					pricing={24.99}
					storage={1}
					memory={10}
					users={5}
					theme={'primary'}
					plane={plan}
				/>
				<Card
					title={'Master'}
					pricing={39.99}
					storage={2}
					memory={20}
					users={10}
					theme={'secundary'}
					plane={plan}
				/>
			</div>
		</div>
	);
};
