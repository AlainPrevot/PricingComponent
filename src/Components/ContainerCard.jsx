import PropTypes from 'prop-types';
import { Card } from '../Components';
import '../Sass/ContainerCard.sass';
import data from '../Data/Data.js';

export const ContainerCard = ({ plan }) => {
	return (
		<div className='container--card'>
			{data.map(data => (
				<Card
					key={data.id}
					title={data.title}
					pricing={data.pricing}
					storage={data.storage}
					memory={data.memory}
					users={data.users}
					theme={data.theme}
					plane={plan}
				/>
			))}
		</div>
	);
};

ContainerCard.propTypes = {
	plan: PropTypes.string.isRequired,
};
