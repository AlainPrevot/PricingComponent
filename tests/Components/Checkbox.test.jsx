import { fireEvent, render, screen } from '@testing-library/react';
import { Checkbox } from '../../src/Components/Checkbox';

describe('Pruebas en Checkbox', () => {
	test('Debe de hacer match con el snapshot', () => {
		const { container } = render(<Checkbox onMode={() => {}} />);

		expect(container).toMatchSnapshot();
	});

	test('Debe de pulsarse el input 2 veces', () => {
		const handleClick = jest.fn();

		render(<Checkbox onMode={handleClick} />);

		const input = screen.getByRole('checkbox');

		fireEvent.click(input);

		fireEvent.click(input);

		expect(handleClick).toHaveBeenCalledTimes(2);
	});

	test('El valor del checkbox debe cambiar al ser pulsado', () => {
		render(<Checkbox onMode={() => {}} />);

		const input = screen.getByRole('checkbox');

		fireEvent.click(input);

		expect(input.checked).toBe(true);

		fireEvent.click(input);

		expect(input.checked).toBe(false);
	});
});
