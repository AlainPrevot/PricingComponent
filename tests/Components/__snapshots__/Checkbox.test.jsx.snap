// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Checkbox Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="container--checkbox"
  >
    <span>
      Annually
    </span>
    <div
      class="checkbox"
    >
      <input
        id="checkbox"
        type="checkbox"
      />
      <label
        for="checkbox"
      />
    </div>
    <span>
      Monthly
    </span>
  </div>
</div>
`;
